includes/wcpdf-functions.php

```diff
@@ -721,3 +721,202 @@ function wpo_wcpdf_get_multilingual_languages(): array {
 	return apply_filters( 'wpo_wcpdf_multilingual_languages', $languages );
 }
 
+/**
+ * Get image mime type
+ *
+ * @param string $src
+ * @return string
+ */
+function wpo_wcpdf_get_image_mime_type( string $src ): string {
+	$mime_type = '';
+
+	if ( empty( $src ) ) {
+		return $mime_type;
+	}
+
+	// Check if 'getimagesize' function exists and try to get mime type for local files
+	if ( function_exists( 'getimagesize' ) && ! filter_var( $src, FILTER_VALIDATE_URL ) ) {
+		$image_info = @getimagesize( $src );
+
+		if ( $image_info && isset( $image_info['mime'] ) ) {
+			$mime_type = $image_info['mime'];
+		} else {
+			wcpdf_log_error( 'getimagesize failed for file: ' . $src );
+		}
+	}
+
+	// Fallback to 'finfo_file' if mime type is empty for local files only (no remote files allowed)
+	if ( empty( $mime_type ) && function_exists( 'finfo_open' ) && ! filter_var( $src, FILTER_VALIDATE_URL ) ) {
+		$finfo = finfo_open( FILEINFO_MIME_TYPE );
+
+		if ( $finfo ) {
+			$mime_type = finfo_file( $finfo, $src );
+			finfo_close( $finfo );
+		} else {
+			wcpdf_log_error( 'Fileinfo failed to open for file: ' . $src );
+		}
+	}
+
+	// Handle remote files
+	if ( empty( $mime_type ) && filter_var( $src, FILTER_VALIDATE_URL ) ) {
```

Comment:
I have replace the function here but because `wp_get_attachment_image_url()` does the same thing we were doing before, so the output should be the same. Maybe you have issues in your paths configuration?

Reply:
Yes, it's still the same.

> Maybe you have issues in your paths configuration?

What do you think the path should be? As far as I know, it should be as a URL instead of an absolute file path when viewing HTML, right?

Reply:
Correct, the public URL.